### ISSUE TITLE
[FEATURE] Add support for while statements 

### ISSUE THREAD

**Issue by TheGupta2012** [qBraid/pyqasm#7]
### Feature Description

Add support for the [`while` loops in openqasm](https://openqasm.com/language/classical.html#while-loops). The main motivation behind this feature is to provide complete semantic coverage of the [openqasm3 grammar](https://openqasm.com/language/index.html)

### Current 
The following code containing a `while` loop in openqasm fails during `unroll` - 

```python
import pyqasm

qasm_str = """
OPENQASM 3.0;
include "stdgates.inc";
qubit q;
bit result;

int i = 0;
// Keep applying hadamards and measuring a qubit
// until 10, |1>s are measured
while (i < 10) {
    h q;
    result = measure q;
    if (result) {
        i += 1;
    }
}"""

module = pyqasm.loads(qasm_str)

module.unroll()
```

```output 
...
ValidationError: Unsupported statement of type <class 'openqasm3.ast.WhileLoop'>
```

### Implementation (Optional)

- Similar to the handling of [`for` loop statements](https://github.com/qBraid/pyqasm/blob/f906f8e0fe22217b03b841a717aa804f9e492279/src/pyqasm/visitor.py#L1595), we will need to add implementation for the [`_visit_while_loop`](https://github.com/qBraid/pyqasm/blob/f906f8e0fe22217b03b841a717aa804f9e492279/src/pyqasm/visitor.py#L1781) method in the [`QasmVisitor` class](https://github.com/qBraid/pyqasm/blob/f906f8e0fe22217b03b841a717aa804f9e492279/src/pyqasm/visitor.py#L50)

- This will require analyzing the structure of [`openqasm3.ast.WhileLoop`](https://github.com/openqasm/openqasm/blob/main/source/openqasm/openqasm3/ast.py#L988) class and figuring out how to evaluate the conditions in different iterations.

**Comment by ryanhill1** [comment]
@atharva-satpute An infinite loop should cause a `ValidationError`. And yes, the solution support both `break` and `continue` statements.

**Comment by ryanhill1** [comment]
Yeah, perhaps an easier way to handle this would be to implement a “maximum loop limit” threshold—similar to how Python raises a `RecursionError: maximum recursion depth exceeded`. The idea would be to set a default threshold, say 1000 iterations, and raise something like a `LoopLimitError` if this limit is exceeded. We could also make this threshold configurable, much like Python’s recursion limit:

```python
import sys

sys.setrecursionlimit(2000)  # Example: Increase recursion limit to 2000
```

Similarly, in our case:

```python
import pyqasm

pyqasm.unroll(..., loop_limit=2000)
```

We could also support `loop_limit=None` to disable this limit entirely, allowing the unrolling process to continue until it hits a `MemoryError` if it exhausts all available memory.

For this initial implementation, it might make sense to simplify things by not tracking “infinite” while loops—just let those cases proceed until a `MemoryError` occurs. We could then spin off a separate issue to add the `loop_limit` feature later.

Maybe @TheGupta2012 can weigh in with his thoughts as well.

**Comment by TheGupta2012** [comment]
@bhagyasreey can you please comment on the issue so that I can assign it to you? This is needed for unitary hack bounty distribution

**Comment by bhagyasreey** [comment]
I would love to work further on the issue. Thank you.
